homeassistant/components/plugwise/number.py

```diff
@@ -0,0 +1,103 @@
+"""Number platform for Plugwise integration."""
+from __future__ import annotations
+
+from collections.abc import Awaitable, Callable
+from dataclasses import dataclass
+from typing import Any
+
+from plugwise import Smile
+
+from homeassistant.components.number import (
+    NumberDeviceClass,
+    NumberEntity,
+    NumberEntityDescription,
+    NumberMode,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import PlugwiseDataUpdateCoordinator
+from .entity import PlugwiseEntity
+
+
+@dataclass
+class PlugwiseEntityDescriptionMixin:
+    """Mixin values for Plugwse entities."""
+
+    command: Callable[[Smile, float], Awaitable[Any]]
+
+
+@dataclass
+class PlugwiseNumberEntityDescription(
+    NumberEntityDescription, PlugwiseEntityDescriptionMixin
+):
+    """Class describing Plugwise Number entities."""
+
+
+NUMBER_TYPES = (
+    PlugwiseNumberEntityDescription(
+        key="maximum_boiler_temperature",
+        command=lambda api, value: api.set_max_boiler_temperature(value),
+        device_class=NumberDeviceClass.TEMPERATURE,
+        name="Maximum Boiler Temperature Setpoint",
+        icon="mdi:thermometer",
```

Comment:
I think the icon is not necessary if there is a device class.
Also, should there be a unit of measurement?

Reply:
Correct, I've  copy-pasted too much from select.py :)
Will fix it later.

Reply:
Suggestions have been implemented.